rpc/wrtc_client.go

```diff
@@ -365,41 +372,29 @@ func dialWebRTC(
 		}
 	}
 
-	utils.PanicCapturingGoWithCallback(func() {
+	utils.PanicCapturingGo(func() {
 		if err := exchangeCandidates(); err != nil {
-			sendErr(err)
+			logger.Warn("Failed to exchange candidates", "err", err)
 		}
-	}, func(err interface{}) {
-		sendErr(fmt.Errorf("%v", err))
 	})
 
-	doCall := func() error {
-		select {
-		case <-exchangeCtx.Done():
-			return multierr.Combine(exchangeCtx.Err(), clientCh.Close())
-		case <-clientCh.Ready():
-			return nil
-		case err := <-errCh:
-			return multierr.Combine(err, clientCh.Close())
-		}
-	}
-
-	if callErr := doCall(); callErr != nil {
-		var err error
-		sendDoneErrorOnce.Do(func() {
+	select {
+	case <-clientCh.Ready():
+		// Happy path
+		sendDone()
+		successful = true
+	case <-exchangeCtx.Done():
+		sendDoneOnce.Do(func() {
 			_, err = signalingClient.CallUpdate(exchangeCtx, &webrtcpb.CallUpdateRequest{
 				Uuid: uuid,
 				Update: &webrtcpb.CallUpdateRequest_Error{
-					Error: ErrorToStatus(callErr).Proto(),
+					Error: ErrorToStatus(exchangeCtx.Err()).Proto(),
```

Comment:
With your changes here, the only error that a caller can send back the signaling server (and potentially to the answerer through the signaling server) is a context canceled or deadline exceeded from the `exchangeCtx`? That makes sense to me but wanted to clarify.

Reply:
> "Thanks for picking up the phone! I just wanted to say I'm calling off the wedding because you didn't pick up the phone."

😆